src/app/actions/main-weather-actions/actions.tsx

```diff
@@ -0,0 +1,28 @@
+import { RootObject } from "../../contracts/weather";
+
+export class SubmitActionFailed { }
+export class SubmitActionPending { }
+export class SubmitActionSucceeded {
+    constructor(private data: RootObject) {
```

Comment:
What kind of thing is `RootObject` ? Who is `data`? Rename both to suitable name.